f5_openstack_agent/lbaasv2/drivers/bigip/service_adapter.py

```diff
@@ -907,3 +913,28 @@ def _apply_esd(self, vip, esd):
             for policy in esd['lbaas_policy']:
                 policies.append({'name': policy, 'partition': 'Common'})
             vip['policies'] = policies
+
+    def _add_listener_bwcPolicy(self, loadbalancer, vip):
+        if 'bandwidth' not in loadbalancer.keys():
+            return
+
+        bwc_mapping = {
+            200: 'low',
+            500: 'mid',
+            1000: 'high'
+        }
+
+        bwc = loadbalancer.get('bandwidth')
+        if bwc not in bwc_mapping.keys():
```

Comment:
If the value of bwc is None, and 'bwcPolicy' string is not in the listener, then all the vip['bwcPolicy']  will be set to 'low' instead of setting none. 

It is designed on purpose right?

Reply:
The bandwidth field in DB is an integer type. Suppose it will never be None. 
However, to be safe for sure, I just added None conditional judgment as you suggested.